src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/FunctionalTestBase.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.pip.data.management.utils;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.pip.data.management.Application;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.restassured.RestAssured.given;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+
+@SpringBootTest(classes = {Application.class, OAuthClient.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@SuppressWarnings("PMD.JUnit5TestShouldBePackagePrivate")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class FunctionalTestBase {
+
+    protected static final String CONTENT_TYPE_VALUE = "application/json";
+
+    @Autowired
+    private OAuthClient authClient;
+
+    protected String accessToken;
+
+    @Value("${TEST_URL:http://localhost:8090}")
+    private String testUrl;
+
+    @BeforeAll
+    void setUp() {
+        RestAssured.baseURI = testUrl;
+        accessToken = authClient.generateAccessToken();
+    }
+
+    protected Response doGetRequest(final String path, final Map<String, String> additionalHeaders) {
+        return given()
+            .relaxedHTTPSValidation()
+            .headers(getRequestHeaders(additionalHeaders))
+            .when()
+            .get(path)
+            .thenReturn();
+    }
+
+    protected Response doPostRequest(final String path, final Map<String, String> additionalHeaders,
+                                     final String body) {
+        return given()
+            .relaxedHTTPSValidation()
+            .headers(getRequestHeaders(additionalHeaders))
+            .body(body)
+            .when()
+            .post(path)
+            .thenReturn();
+    }
+
+    protected Response doDeleteRequest(final String path, final Map<String, String> additionalHeaders,
+                                       final String body) {
+        return given()
+            .relaxedHTTPSValidation()
+            .headers(getRequestHeaders(additionalHeaders))
+            .body(body)
+            .when()
+            .delete(path)
+            .thenReturn();
+    }
+
+    private static Map<String, String> getRequestHeaders(final Map<String, String> additionalHeaders) {
```

Comment:
I don't believe this method is needed / not sure we will need it in the future.

Do you agree and if so can this be removed?

Reply:
This is used by other public method within the class